### DISCUSSION TITLE
Broken BigCommerce Integration

### DISCUSSION THREAD

**Discussion by brightonmike** [vercel/community#362] [Help] [answered]
Hello.

When I try and setup the BigCommerce integration, the callback for the popover window triggers a 500 error.

If I close the popup and proceed, the integration isn't listed in Vercel and the site will not connect to my BigCommerce account. However, the app will then show up against the store.

I've had to delete the app, the repo and the Vercel site several times in an attempt to get this working but with no luck. I've also tried logging out of BigCommerce, and using incognito. 

If anyone has any ideas be great.

<img width="1104" alt="Screenshot 2022-02-15 at 15 08 26" src="https://user-images.githubusercontent.com/2071150/154098279-478791ba-db06-4282-91d0-cd32c4c48f2a.png">

**Comment by swarnava** [comment, marked as answer]
Hey there,

If you can check your project settings, as long as these environment variables are set, there’s no need to get it connected to the BigCommerce integration.

```
    BIGCOMMERCE_CHANNEL_ID
    BIGCOMMERCE_STOREFRONT_API_URL
    BIGCOMMERCE_STOREFRONT_API_TOKEN
    BIGCOMMERCE_STORE_API_URL
    BIGCOMMERCE_STORE_API_TOKEN
    BIGCOMMERCE_STORE_API_CLIENT_ID
```

These environment variables can be set manually without using the BigCommerce integration by creating a developer account on BigCommerce and generating API tokens (STORE_API_… are REST tokens and STOREFRONT_API_… are GraphQL tokens
